src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -61,11 +61,12 @@ public function denormalize($data, $type, $format = null, array $context = [])
             throw NotNormalizableValueException::createForUnexpectedDataType('The data is neither an integer nor a string, you should pass an integer or a string that can be parsed as an enumeration case of type '.$type.'.', $data, [Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true);
         }
 
-        try {
-            return $type::from($data);
-        } catch (\ValueError $e) {
-            throw NotNormalizableValueException::createForUnexpectedDataType($e->getMessage(), $data, [Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true, $e->getCode(), $e);
+        $matched = $type::tryfrom($data);
+        if (!$matched) {
+            throw new InvalidArgumentException('The data must belong to a backed enumeration.');
```

Comment:
The exception message could probably give more hints about what is happening, starting with which backed enum?

Reply:
The `ValueError` that was caught previously contained all that information. Switching to `tryForm()` does not look like a good idea, tbh.

Reply:
Thanks for your feedback! 

it was not possible to collect an PartialDenormalizationException. The error message received was TypeError: DummyObject::__construct(): Argument #1 ($test) must be of type TestEnum, string given, which seemed not very consistent compared to what was really the error. 

It appears that it cannot be created because there is no matching (property or value)

Reply:
This proposal is mainly to avoid the TypeError when DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true

Reply:
Yeah, I got that. But as far as I understood – please correct me if I got that wrong – you're attempting to fix the bug by changing the exception that we raise when the enum cannot be hydrated. But you are also changing the way we detect that the enum cannot be hydrated. Those are two independent changes and I don't see why the latter is necessary.

Reply:
In fact I think it's the test that is quite misleading ( Serializer::testCollectDenormalizationErrorsWithEnumConstructor()
) because it does not change the behavior whether it's used with 'from' or 'tryFrom' (the result is the same). 
it's the case that is tested and not the scalar equivalent.  => '{"invalid": "get"}' 

it's more about the BackedEnumNormalizerTest::testDenormalizeBadBackingValueThrowsException => with a different exception raised  for '{"get": "invalid"}' and avoid a TypeError. I could in the message error add some details

Reply:
> it does not change the behavior whether it's used with 'from' or 'tryFrom' (the result is the same).

So, we can revert back to `from()`?

Reply:
Maybe I'm wrong, but what seems annoying is the TypeError in some cases with Argument #1 ($get) must be of type Symfony\Component\Serializer\Tests\Fixtures\StringBackedEnumDummy, string given" message

`try {`
           ` $serializer->deserialize('{"get": "invalid"}', DummyObject::class, 'json', [`
               ` DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true`
            `]);`
       ` } catch (\Throwable $th) {`
          `$this->assertInstanceOf(TypeError::class, $th);`
        `}`

if we use tryFrom we catch an InvalidArgumentException with "The data must belong to a backed enumeration of type TestEnum" as message

Reply:
I'm sorry then. 

If I may, 
if we add an other backedEnum with TEST="invalid" (e.g  StringBackedEnumDummy2) . but in the constructor of DummyObject it still aims for StringBackedEnumDummy, the test will fail but not because it's a string, but because the data doesn't belong to a backed enumeration of type StringBackedEnumDummy (belongs to StringBackedEnumDummy2)

thanks again for your feedback 🙏

Reply:
Okay, let's try again. Please try to focus on my question and try to answer that question only.

Why did you change the `from()` call to `tryFrom()`. Why is this change necessary to achieve your goal?

Reply:
Ok, I try an other approach. And don't worry I'm focus on your question :)

With "from" and how ValueError is handled: 

even if we know that the value doesn't match (BackedEnumNormalizer throw ValueError `$type::from($data)), AbstractNormalizer continue and call (line 419) $reflectionClass->newInstanceArgs($params). So it will try to create a new instance with an argument we know doesn't match.  And then we receive a TypeError with "__construct(): Argument #1 ($get) must be of type ......., string given"

With "tryFrom" : handle null (in case of failure) and throw directly an InvalidArgumentException. 

But, it's true we could do the same thing with "from" if we throw an InvalidArgumentException instead of  throwing NotNormalizableValueException::createForUnexpectedDataType(.....)

Reply:
The problem is when DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true is set

Reply:
Seems like the change was reverted.